GAE/test/org/akvo/flow/api/app/DataPointUtilTest.java

```diff
@@ -0,0 +1,190 @@
+/*
+ *  Copyright (C) 2020 Stichting Akvo (Akvo Foundation)
+ *
+ *  This file is part of Akvo FLOW.
+ *
+ *  Akvo FLOW is free software: you can redistribute it and modify it under the terms of
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.flow.api.app;
+
+import com.gallatinsystems.device.dao.DeviceDAO;
+import com.gallatinsystems.device.domain.Device;
+import com.gallatinsystems.framework.domain.BaseDomain;
+import com.gallatinsystems.survey.dao.SurveyDAO;
+import com.gallatinsystems.survey.domain.Survey;
+import com.gallatinsystems.surveyal.dao.SurveyedLocaleDao;
+import com.gallatinsystems.surveyal.domain.SurveyedLocale;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.akvo.flow.dao.SurveyAssignmentDao;
+import org.akvo.flow.domain.persistent.SurveyAssignment;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.waterforpeople.mapping.dao.SurveyInstanceDAO;
+import org.waterforpeople.mapping.domain.SurveyInstance;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DataPointUtilTest {
+    private final static LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    @BeforeAll
+    public static void setUp() {
+        helper.setUp();
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        helper.tearDown();
+    }
+
+    private long randomId() {
+        return ThreadLocalRandom.current().nextLong(0, 1000000);
+    }
+
+    private int randomInt() {
+        return ThreadLocalRandom.current().nextInt(1, 200);
+    }
+
+    private Survey createForm() {
+        long id = randomId();
+        Survey form = new Survey();
+        form.setKey(KeyFactory.createKey("Survey", id));
+        form.setName("Form" + id);
+        form.setVersion(1.0d);
+        SurveyDAO dao = new SurveyDAO();
+        return dao.save(form);
+    }
+
+    private List<SurveyInstance> createFormInstances(Survey form, SurveyedLocale survey, int howMany) {
```

Comment:
Please rename: Survey is called form and SurveyedLocale is called survey, it's a bit confusing

Reply:
Ok I see below SurveyedLocale is named dataPoint, maybe just rename that

Reply:
Unfortunately this is the mental burden a Flow developer needs to deal with:
* Domain -> Entity
  * Survey -> SurveyGroup
  * Datapoint -> SurveyedLocale
  * Form -> Survey